src/utransport.rs

```diff
@@ -12,13 +12,37 @@
  ********************************************************************************/
 
 use async_trait::async_trait;
+use std::hash::{Hash, Hasher};
+use std::ops::Deref;
+use std::sync::Arc;
 
 use crate::{UMessage, UStatus, UUri};
 
-/// `UTransport` is the uP-L1 interface that provides a common API for uE developers to send and receive messages.
+/// `UListener` is the uP-L1 interface that provides a means to create listeners which are registered to `UTransport`
 ///
-/// Implementations of `UTransport` contain the details for connecting to the underlying transport technology and
-/// sending `UMessage` using the configured technology. For more information, please refer to
+/// Implementations of `UListener` contain the details for what should occur when a message is received
+/// For more information, please refer to
+/// [uProtocol Specification](https://github.com/eclipse-uprotocol/uprotocol-spec/blob/main/up-l1/README.adoc).
+pub trait UListener: 'static + Send + Sync {
```

Comment:
- [ ] Based on the feedback today in the uProtocol Zenoh meeting, I need to mark the `on_receive()` and `on_error()` functions as async to allow the `up-client-foo-rust` to asynchronously call these functions so we don't block that thread.

@AnotherDaniel, @sophokles73, @tamarafischer, @stevenhartley -- did I capture the discussion correctly? WDYT?

Reply:
Yes that sounds like a useful thing to do in this context.

Reply:
I don't believe forcing the trait to be `async` actually solves the problem, although it may mitigate it a bit. If the user still performs long synchronous computation inside the callback, the thread will still be blocked. The choice here is more related to internal implementation and to potentially dedicated a thread pool to the user different from the one used by the middleware.

Reply:
I agree @Mallets, this is just more of enabling the `UTransport` implementation to use a thread pool.

Perhaps we should add some additional comments above `on_receive()` and `on_error()` to clarify that these should be assumed to execute quickly and return.

WDYT?